### ISSUE TITLE
Error with exec credentials on the latest kubernetes-model

### ISSUE THREAD

**Issue by janario** [fabric8io/kubernetes-client#1332]
After update the kubernetes-client from `4.1.0` to `4.1.1` the exec credentials (used by aws eks) is not working

It used to get it from the map of other properties https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java#L511

But on the latest version o kubernetes-model, AuthInfo, `exec` is well mapped and additionalProperties doesn't have it anymore

**Comment by rohanKanojia** [comment]
@janario : What error are you facing? It could be because of #1280

**Comment by janario** [comment]
The exception:
```
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: <url>/api/v1/pods. Message: Forbidden! User <roleArn> doesn't have permission. pods is forbidden: User "system:anonymous" cannot list pods at the cluster scope.
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:478) ~[kubernetes-client-4.1.1.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:415) ~[kubernetes-client-4.1.1.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:381) ~[kubernetes-client-4.1.1.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:344) ~[kubernetes-client-4.1.1.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:328) ~[kubernetes-client-4.1.1.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.listRequestHelper(BaseOperation.java:193) ~[kubernetes-client-4.1.1.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:618) ~[kubernetes-client-4.1.1.jar:na]
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:68) ~[kubernetes-client-4.1.1.jar:na]
	at com.raisin.KubeLogsConsumer.run(KubeLogsConsumer.java:30) ~[classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_181]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_181]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_181]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_181]
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:343) ~[spring-aop-5.1.4.RELEASE.jar:5.1.4.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198) ~[spring-aop-5.1.4.RELEASE.jar:5.1.4.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.1.4.RELEASE.jar:5.1.4.RELEASE]
	at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:115) ~[spring-aop-5.1.4.RELEASE.jar:5.1.4.RELEASE]
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266) ~[na:1.8.0_181]
	at java.util.concurrent.FutureTask.run(FutureTask.java) ~[na:1.8.0_181]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_181]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_181]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_181]

```


My exec user is something like:
```
- name: <arn>
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - token
      - -i
      - <cluster>
      command: aws-iam-authenticator
      env:
      - name: AWS_PROFILE
        value: <profile>
```

And I'm doing something simple like `PodList list = client.pods().inNamespace("janario").list();`

As I debug it https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/Config.java#L511

currentAuthInfo.getAdditionalProperties().get("exec"); is returning null because `exec` used to be mapped as `JsonAnySetter` for other properties

I think it should be changed to something like `ExecConfig exec = currentAuthInfo.getExec()` and generate the token

ExecConfig has all the required fields command, env...

**Comment by antonalechnovic** [comment]
lol having the same issue, funny thought when you downgrade to 4.1.0 you get :
Unrecognized field "expirationTimestamp"

Not sure what to do :-D

**Comment by janario** [comment]
4.1.0 is working fine for me, We don't have `expirationTimestamp` in our json (not sure why, maybe our iam configs or the version of aws-iam-authenticator). Anyway that was fixed at #1280 

---

About this one, I've made a pull request and I re-enabled a test-case, please @rohanKanojia could you take a look? :-)
https://github.com/fabric8io/kubernetes-client/pull/1333

Thanks guys

btw, it is an awesome library, I really like it ;-) this was just a small bug
